tests/unittest/_torch/ray_orchestrator/multi_gpu/test_accuracy_with_allreduce_strategy.py

```diff
@@ -0,0 +1,386 @@
+            # Extract logprob values from the list of dicts
+            logprob_values = [
+                logprobs[token_id].logprob for token_id, logprobs in zip(token_ids, logprobs_list)
+            ]
+            llm_responses.append(token_ids)
+            llm_logprobs.append(torch.tensor(logprob_values, dtype=torch.float32, device="cuda"))
+    finally:
+        if ray.is_initialized():
+            ray.shutdown()
+
+    torch.cuda.empty_cache()
+    input_ids, attention_mask, position_ids = pad_data(test_prompts, llm_responses)
+
+    # Split data across GPUs
+    num_gpus = 4
+    micro_batch_size = 16
+    batch_size = input_ids.shape[0]
+    samples_per_gpu = (batch_size + num_gpus - 1) // num_gpus
+
+    dp_hf_models = []
+    for device_id in range(num_gpus):
+        hf_model = HFModel(model_dir, device_id)
+        dp_hf_models.append(hf_model)
+
+    # Split input data and responses into chunks for each GPU
+    input_ids_chunks = []
+    attention_mask_chunks = []
+    position_ids_chunks = []
+    responses_chunks = []
+
+    for i in range(num_gpus):
+        start_idx = i * samples_per_gpu
+        end_idx = min((i + 1) * samples_per_gpu, batch_size)
+
+        if start_idx < batch_size:
+            input_ids_chunks.append(input_ids[start_idx:end_idx])
+            attention_mask_chunks.append(attention_mask[start_idx:end_idx])
+            position_ids_chunks.append(position_ids[start_idx:end_idx])
+            responses_chunks.append(llm_responses[start_idx:end_idx])
+
+    # Process each chunk on its corresponding GPU asynchronously
+    async def process_all_chunks():
+        tasks = []
+        for i, (input_chunk, attn_chunk, pos_chunk, resp_chunk) in enumerate(
+            zip(input_ids_chunks, attention_mask_chunks, position_ids_chunks, responses_chunks)
+        ):
+            task = generate_batch_async(
+                dp_hf_models[i],
+                input_chunk,
+                attn_chunk,
+                pos_chunk,
+                resp_chunk,
+                prompt_max_len=1024,
+                micro_batch_size=micro_batch_size,
+            )
+            tasks.append(task)
+        return await asyncio.gather(*tasks)
+
+    ref_logprobs_chunks = asyncio.run(process_all_chunks())
+
+    # Move all tensors to cuda:0 and flatten the list
+    # Each GPU returns a list of logprobs tensors
+    ref_new_token_logprobs = []
+    for i, logprobs_list in enumerate(ref_logprobs_chunks):
+        for logprobs in logprobs_list:
+            ref_new_token_logprobs.append(logprobs.to("cuda:0"))
+
+    assert len(ref_new_token_logprobs) == batch_size, (
+        f"Count mismatch: got {len(ref_new_token_logprobs)}, expected {batch_size}"
+    )
+
+    del dp_hf_models
+    torch.cuda.empty_cache()
+
+    # Compare LLM logprobs vs HF reference
+    if allreduce_strategy == "AUTO":
+        with pytest.raises(AssertionError, match=r"Final Min diff: .* is below threshold -2\.30"):
```

Comment:
Hi @shuyixiong , instead of doing this, can we file an NVBug to capture the TRTLLM AllReduce kernel issue if we haven't, and pytest skip this test with the NVBug id when parameterized with `AUTO` ?